### DISCUSSION TITLE
Creation of complex data types of the entity IFCSITE and IFCRIGIDOPERATION in IDS

### DISCUSSION THREAD

**Discussion by glunzm** [buildingSMART/IDS#387] [Q&A]
Hello,
As part of the investigation into how georeferencing can be implemented using IDS and IFC, a problem has been identified in the definition of attributes of the IFCSITE and IFCRIGIDOPERATION entity (in relation to IFCGEOGRAPHICCRS). When checking the IDS, there is no valid match of data types between the values of the attributes REFLATIDUDE (IFCSITE) and REFLONGITUDE (IFCSITE) as well as FIRSTCOORDINATE (IFCRIGIDOPERATION) and SECONDCOORDINATE (IFCRIGIDOPERATION) defined in the IDS document.

Definitions of the IFC classes:
<img width="604" alt="24-11-18-IFCRIGIDOPERATION-Attribute" src="https://github.com/user-attachments/assets/4b4624eb-2795-4e66-8bdb-a7bb23cabdf4">
<img width="603" alt="24-11-18-IFCSITE-Attribute" src="https://github.com/user-attachments/assets/ae53af18-f349-4144-8400-bf90e221fed9">

Example of the error message:
<img width="922" alt="24-11-18-IFCSITE" src="https://github.com/user-attachments/assets/892e7240-0491-426b-af7c-d286d188aa65">

My assumption is that individual values based on complex data types cannot be implemented or defined and checked by IDS.
Is there a way to make it possible to check complex data types? 

Thank you in advance!

**Comment by atomczak** [comment]
It looks to me that this is a problem with your UPPERCASE. The attributes should be in PascalCase, so RefLatitude and RefLongitude. If that's not it, please share the IDS file here.

**Comment by glunzm** [comment]
Thanks, the UpperCase was an issue, but not all of it. My question is:
How do I define a restriction for the values to be exactly -87.35.40 (Chicago Harbour as in IfcSite example) for IfcSite according to IfcCompoundPlaneAngleMeasure (degrees, minutes, seconds)?
https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcSite.htm
Happy about any help!

[20241126_IfcSite-RefLongitude-question.ids.txt](https://github.com/user-attachments/files/17932486/20241126_IfcSite-RefLongitude-question.ids.txt)
<img width="910" alt="24-11-22-RefLongitude" src="https://github.com/user-attachments/assets/8b14edc1-c4fd-4be2-8fb7-5e8cb94eb0ed">

**Reply by Moult** [reply]
You cannot. IDS does not support compound data like this, only basic data types.

You can however specify a rigid operation / map conversion / etc with specific eastings and northings. If georeferencing is your goal, that is the appropriate place, not "RefLong/lat".

**Reply by aothms** [reply]
We discussed this quite extensively btw in private discussions, whether RefLong/lat as a DMS value could be mapped to a single decimal float for the purpose of IDS. But it would be a fairly one-off thing for an 'almost deprecated' attribute.
